port_drayage_plugin/src/port_drayage_worker.cpp

```diff
@@ -82,8 +82,28 @@ namespace port_drayage_plugin
         using boost::property_tree::ptree;
         ptree pt;
         pt.put("cmv_id", _cmv_id);
-        pt.put("cargo_id", _cargo_id);
         pt.put("operation", PORT_DRAYAGE_ARRIVAL_OPERATION_ID);
+
+        // Add current vehicle location (latitude and longitude)
+        ptree location;
+        location.put("latitude", _current_gps_position.latitude * 10000000); // Convert degrees to 1/10 microdegrees
```

Comment:
Is there a particular reason this is being read straight from the GPS itself (based on the INSPVA message)? Fundamentally that should work fine, mostly. But I think the more idiomatic approach here would involve still subscribing to the final output of localization, but then transforming it into lat/lon via the proj library.

Reply:
I've fixed this to follow the approach used in other plugins.